partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprUnionAll.kt

```diff
@@ -0,0 +1,24 @@
+package org.partiql.eval.internal.operator.rex
+
+import org.partiql.eval.internal.Environment
+import org.partiql.eval.internal.helpers.IteratorChain
+import org.partiql.eval.internal.helpers.asIterator
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.eval.value.Datum
+
+internal class ExprUnionAll(
+    private val lhs: Operator.Expr,
+    private val rhs: Operator.Expr,
+    private val coerce: Boolean,
```

Comment:
IMO, there should be as little logic in the execution nodes as possible, unless absolutely necessary. So, for `ExprUnionAll`, maybe there are two implementations: `ExprUnionAll.Permissive` and `ExprUnionAll.Strict`. That way, there is no additional planning logic in the actual execution.

Same goes for the other nodes.

Reply:
Latest commit removes these `Rex` exprs and defers to use the existing `Rel` scan logic.